### PR TITLE
[Zero-Dim] support input 0D Tensor for softmax/log_softmax/gumbel_softmax

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

支持softmax/log_softmax/gumbel_softmax输入0D Tensor，新增功能，无不兼容影响
---

```
paddle.nn.functional.softmax
paddle.nn.Softmax

paddle.nn.functional.log_softmax
paddle.nn.LogSoftmax

paddle.nn.functional.gumbel_softmax
paddle.nn.GumbelSoftmax

paddle.randint
```

### CODE REVIEW

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by kolinwei — APPROVED** [review]
